packages/block-editor/src/components/global-styles/dimensions-panel.js

```diff
@@ -183,12 +183,13 @@ function DimensionsToolsPanel( {
 }
 
 const DEFAULT_CONTROLS = {
-	contentSize: true,
-	wideSize: true,
-	padding: true,
-	margin: true,
-	blockGap: true,
-	minHeight: true,
+	contentSize: false,
+	wideSize: false,
+	padding: false,
+	margin: false,
+	blockGap: false,
+	minHeight: false,
```

Comment:
In other words, a user of `DimensionsControl` shouldn't have to define the "defaultControls" prop in order to have a working component.

Reply:
Any thoughts on this? I'm actually not really satisfied that this PR changed the behavior of default controls in one Styles UI component but not the others, I'm also not convinced that this is the right behavior. Global styles are not customizable by the user, so they shouldn't need to pass any "defaultControls" prop, while the "blocks" (block inspector) are the ones that can customize the default controls, so they should be the ones providing a custom value for the defaultControls prop.

Reply:
The main reason I went with this option was to not change things for third party blocks. Previously, the default was none of the controls would display _except_ for global styles, but the default setting for the controls was to display all. Blocks outside of global styles weren't displaying controls because the defaults were being silently overridden by an empty object, which is non-ideal. But that's how things have been for a while, and if we change the default behaviour now, any custom blocks using these block supports will suddenly have a bunch of controls displaying by default that they didn't have before.

Personally, I'd love to have everything display by default, because having to look for controls in dropdowns is excruciatingly painful UX, but I'm concerned about changing things around for extenders with no warning.

I'm happy to make the same change for the other block supports as long as we're clear this is the way we want to go.

Reply:
I agree that it's not idea to change behavior on block authors but maybe we can make some small adjustments as it's not entirely a breaking change, things still work, they may just be hidden or visible. I think a dev note in this case would be cool.

Maybe we should try to make the change with the least disruption to block authors while making sure the default values are decent and consistent across the components. what would that be?

Reply:
> making sure the default values are decent and consistent across the components. what would that be?

I'd be happy to make them all visible by default all the time, but perhaps best defer to @WordPress/gutenberg-design 😄